File: suites/05_ml_primitives/comparisons/softmax.go
package main
import "fmt"
func softmaxLike(width, rounds int64) int64 {
	var total int64
	for r := int64(0); r < rounds; r++ {
		var partial int64
		for l := int64(0); l < width; l++ { partial += ((l * 11) + r) % 251 }
		total += partial
	}
	return total
}
func main() { fmt.Println(softmaxLike(4096, 4000)) }

File: suites/05_ml_primitives/comparisons/softmax.rs
fn softmax_like(width: i64, rounds: i64) -> i64 {
    let mut total: i64 = 0;
    for r in 0..rounds {
        let mut partial: i64 = 0;
        for l in 0..width { partial += ((l * 11) + r) % 251; }
        total += partial;
    }
    total
}
fn main() { println!("{}", softmax_like(4096, 4000)); }
